### PR TITLE
feat(RHINENG-11806): Handle OPTIONAL_REPOSITORIES task parameter

### PR DESCRIPTION
**Pull request by mhuth**
Screenshots showing the Parameter form with the optional repositories checkboxes all checked and the value produced that is sent to the Tasks API:
Screenshot from 2024-08-22 15-39-58
Screenshot from 2024-08-22 15-40-10

More screenshots showing none of the optional repositories checkboxes checked and the value 'None' sent to the backend:
Screenshot from 2024-08-22 15-42-04
Screenshot from 2024-08-22 15-42-14

### CODE REVIEW

File: src/SmartComponents/RunTaskModal/ConversionTaskInputParameters.js [hunk 1]
```diff
@@ -38,6 +44,12 @@ const ConversionTaskInputParameters = ({
   const [skipPackageCheckValue, setSkipPackageCheckValue] = useState(
     skipPackageCheck.default
   );
+  const [optionalRepositoriesValue, setOptionalRepositoriesValue] = useState(
+    optionalRepositories.default === 'None' ||
```
Comment (mkholjuraev) [inline src/SmartComponents/RunTaskModal/ConversionTaskInputParameters.js:48]: How about making this more shorter.

```suggestion
    [ 'None', '' ].includes(optionalRepositories.default)
    ? [] 
    : optionalRepositories.default.split(',')
```

File: src/SmartComponents/RunTaskModal/ConversionTaskInputParameters.js [hunk 2]
```diff
@@ -175,6 +187,37 @@ const ConversionTaskInputParameters = ({
         skipPackageCheckValue,
         setSkipPackageCheckValue
       )}
+
+      <FormGroup role="group" fieldId="optional-repositories">
```
Comment (mkholjuraev) [inline src/SmartComponents/RunTaskModal/ConversionTaskInputParameters.js:191]: I feel this Form can be extracted into a separate file with its own state and logic.

File: src/SmartComponents/RunTaskModal/ConversionTaskInputParameters.js (hunk 1 above)
Comment (mhuth) [reply src/SmartComponents/RunTaskModal/ConversionTaskInputParameters.js]: Thanks for the tip. Done

**Review by mkholjuraev — APPROVED** [review]
LGTM!

**Pull request merged**